### PR TITLE
[etcd-cpp-apiv3] update to 0.15.3

### LINKED ISSUE #32666: [etcd-cpp-apiv3] update to 0.15.2
### Library name

etcd-cpp-apiv3

### New version number

0.15.2

### Other information that may be useful (release notes, etc...)

current version in 0.14.2

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes #32666 

Update port etcd-cpp-apiv3 the latest version 0.15.3
Fix error:
````
fatal error LNK1107: invalid or corrupt file: cannot read at 0x330
````
Fix error :
````
error: use of undeclared identifier 'IPPROTO_TCP'
````
Note: no feature need to test.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/etcd-cpp-apiv3/fix-undeclared-error.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+index d5a4179..1633088 100644
+--- a/src/SyncClient.cpp
++++ b/src/SyncClient.cpp
+@@ -4,6 +4,10 @@
+ #define NOMINMAX
+ #endif
+ 
++if(VCPKG_TARGET_IS_ANDROID)
++#include <netinet/in.h>
++endif()
```
Comment (dg0yt) [inline ports/etcd-cpp-apiv3/fix-undeclared-error.patch]: This is putting cmake script code into a C++ file. This cannot work.

Comment (JonLiu1993) [reply ports/etcd-cpp-apiv3/fix-undeclared-error.patch]: Yes, put it in directly, it cannot be found on windows, I will consider other solutions.

File: ports/etcd-cpp-apiv3/fix-undeclared-error.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+diff --git a/src/SyncClient.cpp b/src/SyncClient.cpp
+index d5a4179..1633088 100644
+--- a/src/SyncClient.cpp
++++ b/src/SyncClient.cpp
+@@ -4,6 +4,10 @@
+ #define NOMINMAX
+ #endif
+ 
++#ifdef __ANDROID__
```
Comment (JonLiu1993) [inline ports/etcd-cpp-apiv3/fix-undeclared-error.patch:9]: Fix error :
````
error: use of undeclared identifier 'IPPROTO_TCP'
````

**Review by BillyONeal — DISMISSED** [review]
Can you submit these patches upstream?

**Comment by JonLiu1993** [comment]
> Can you submit these patches upstream?

Thank you for the reminder, I have submitted a PR [#244](https://github.com/etcd-cpp-apiv3/etcd-cpp-apiv3/pull/244) on Upstream.

**Comment by sighingnow** [comment]
Hi @JonLiu1993 The pull ruquest on etcd-cpp-apiv3 has been merged and I have just cut a new release v0.15.3, please use that version in vcpkg.

Thanks!

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update and following up with upstream!

**Pull request merged**